tools/ports/mstelemetry/portfile.cmake

```diff
@@ -4,45 +4,80 @@ message("CMAKE_CURRENT_SOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}")
-message("Using local source snapshot from ${SOURCE_PATH}")
+    # Use local snapshot since we already cloned the code
+    get_filename_component(SOURCE_PATH "${CMAKE_CURRENT_LIST_DIR}/../../../" ABSOLUTE)
+    message("Using local source snapshot from ${SOURCE_PATH}")
 else()
-# Fetch from GitHub master
-message("Fetching source code from GitHub...")
-vcpkg_from_github(
-    OUT_SOURCE_PATH SOURCE_PATH
-    REPO microsoft/cpp_client_telemetry
-    HEAD_REF master
-)
+    # Fetch from GitHub master
+    message("Fetching source code from GitHub...")
+    if (UNIX)
+        vcpkg_from_github(
+            OUT_SOURCE_PATH SOURCE_PATH
+            REPO microsoft/cpp_client_telemetry
+            HEAD_REF master
+        )
+    else()
+        vcpkg_from_github(
+            OUT_SOURCE_PATH SOURCE_PATH
+            REPO microsoft/cpp_client_telemetry
+            REF 4f60dd3bca305c2c0dd5ec2ed7b91d36b4de6dcf
+            SHA512 9778df5aa65d95fe1d41739753495d29b3149676e98ac2e802a103604553f4f2b43bc2eb089c2e13dc695f70279287ea79ec6e2926fad03befe8a671f91d36fb
+            HEAD_REF master
+            PATCHES ${CMAKE_CURRENT_LIST_DIR}/v142-build.patch
+        )
+    endif()
 endif()
 
 # TODO: it will be slightly cleaner to perform pure CMake or Ninja build, by describing all possible variable options
 # as separate triplets. Since we have a fairly non-trivial build logic in build.sh script - we use it as-is for now.
 # build.sh itself should check if we are building under vcpkg and avoid installing deps that are coming from vcpkg.
 if (UNIX)
-vcpkg_execute_build_process(
-    COMMAND ${SOURCE_PATH}/build.sh noroot
-    WORKING_DIRECTORY ${SOURCE_PATH}/
-    LOGNAME build
-)
-
-vcpkg_execute_build_process(
-    COMMAND ${SOURCE_PATH}/install.sh ${CURRENT_PACKAGES_DIR}
-    WORKING_DIRECTORY ${SOURCE_PATH}/
-    LOGNAME install
-)
+    vcpkg_execute_build_process(
+        COMMAND ${SOURCE_PATH}/build.sh noroot
+        WORKING_DIRECTORY ${SOURCE_PATH}/
+        LOGNAME build
+    )
+
+    vcpkg_execute_build_process(
+        COMMAND ${SOURCE_PATH}/install.sh ${CURRENT_PACKAGES_DIR}
+        WORKING_DIRECTORY ${SOURCE_PATH}/
+        LOGNAME install
+    )
+
+    file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
 else()
-# TODO: verify Windows build
-vcpkg_execute_build_process(
-    COMMAND ${SOURCE_PATH}/build-all.bat
-    WORKING_DIRECTORY ${SOURCE_PATH}/
-    LOGNAME build
-)
-endif()
+    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+    vcpkg_install_msbuild(
+        SOURCE_PATH ${SOURCE_PATH}
+        PROJECT_SUBPATH Solutions/MSTelemetrySDK.sln
+        LICENSE_SUBPATH LICENSE
+        RELEASE_CONFIGURATION Release
+        DEBUG_CONFIGURATION Debug
+        OPTIONS /p:MATSDK_SHARED_LIB=1
+        PLATFORM ${VCPKG_TARGET_ARCHITECTURE}
+        PLATFORM_TOOLSET v142
```

Comment:
I think we can add two options here:
- one is to respect env variable `PLATFORM_TOOLSET` , if it is already set (for those who still use vs2017-vc141)
- another option is to auto-detect the compiler. It's a bit more involved. But I can help with a script for that.

In general, maybe just replacing it with a custom variable logic - if set, use the value from var; else if not set, then assume the tool chain is v142 - would be good enough at first.